src/Units/KaijuWarsWeapons.java

```diff
@@ -0,0 +1,432 @@
+
+    public CannonOfBoom()
+    {
+      super(3, 2, MIN_RANGE, MAX_RANGE);
+    }
+  }
+  public static class GuncrossWing extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public GuncrossWing()
+    {
+      super(2, 3);
+      isAirWeapon = true;
+    }
+  }
+  public static class GuncrossRobot extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public GuncrossRobot()
+    {
+      super(4, 2);
+    }
+  }
+  public static class SuperZ2 extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public SuperZ2()
+    {
+      super(3, 2);
+    }
+  }
+  public static class BigBoy extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public BigBoy()
+    {
+      super(4, 0);
+      isAirWeapon = true;
+    }
+  }
+  public static class Kaputnik extends ShootAll
+  {
+    private static final long serialVersionUID = 1L;
+
+    public Kaputnik()
+    {
+      super(8);
+      isAirWeapon = true;
+    }
+  }
+  public static class SharkJet extends ShootAll
+  {
+    private static final long serialVersionUID = 1L;
+
+    public SharkJet()
+    {
+      super(1);
+      isAirWeapon = true;
+    }
+  }
+
+  // Detailed damage calcs
+
+  public static int deriveAttack(KaijuWarsWeapon gun, KaijuWarsUnitModel defender)
+  {
+    int attack = 0;
+    if( defender.isAirUnit() )
+      attack = gun.vsAir;
+    else
+      attack = gun.vsLand;
+    return attack;
+  }
+  public static int deriveCounter(KaijuWarsWeapon gun, KaijuWarsUnitModel defender)
+  {
+    int counterPower = defender.kaijuCounter;
+    if( !gun.negateCounterBonuses )
+    {
+      if( gun.isAirWeapon && defender.slowsAir )
+        counterPower += SLOW_BONUS;
+      if( !gun.isAirWeapon && defender.slowsLand )
+        counterPower += SLOW_BONUS;
+      if( defender.resistsKaiju )
+        counterPower += SLOW_BONUS;
+    }
+    return counterPower;
+  }
+
+  public static double getDamage(int attack, int counterPower)
+  {
+    return getDamageRatioStyle(attack, counterPower);
+  }
+
+  /**
+   * Produces damage numbers based on attack/kaijuCounter
+   */
+  public static double getDamageRatioStyle(int attack, int counterPower)
+  {
+    // 1-based instead of 0-based
+    int durability = 1 + counterPower;
+    int damage = attack * 1000 / durability;
+    // Round properly
+    damage += 5;
+    damage /= 10;
+
+    return damage * KAIJU_DAMAGE_FACTOR / 100;
+  }
+
+  /**
+   * Produces damage numbers centered around KAIJU_DAMAGE_BASE
+   */
+  public static double getDamageShiftingStyle(int attack, int durability)
+  {
+    int damage = KAIJU_DAMAGE_BASE;
+
+    int finalPower = attack - durability;
+    if( finalPower > -3 )
+      damage += finalPower * 10;
+    else if( finalPower > -8 )
+      damage = damage - 10 + finalPower * 5;
+    else
+      damage = 2 * (12 + finalPower);
+
+    return damage;
+  }
```

Comment:
This is the simpler version? It's made out of edge cases.

Reply:
Good point. I don't actually remember what I was doing with that, I guess.

Reply:
For future reference (have added comments, but they're not visible here):
The intent was relatively consistent addition/subtraction from a base value - the difference matters, not how big each side is.
The edge cases are to make it decay to 0 less abruptly:
55 is the base value. Any +1 in KW = +10 here.
The first two -1s give -10
The next 5 give -5
Any further give -2 (and I didn't lower-bound since the lowest difference is -8)